digits/scheduler.py

```diff
@@ -175,7 +176,11 @@ def delete_job(self, job):
                     for j in self.jobs:
                         if isinstance(j, ModelJob) and j.dataset_id == job.id():
                             logger.error('Cannot delete %s (%s) because %s (%s) depends on it.' % (job.name(), job.id(), j.name(), j.id()))
-                            return False
+                            DependencyExists = True
+                            ErrorDetails += j.name() + ','
+                if DependencyExists:
+                    ErrorDetails = 'Cannot delete %s because jobs - %s depends on it' % (job.name(), ErrorDetails)
+                    raise errors.DeleteError(ErrorDetails)
```

Comment:
How about this message instead:

    Cannot delete "<dataset_name>" because <count> models depend on it: "<model1>", "<model2>", "<model3>"...